### PR TITLE
Debug test `01600_parts_states_metrics_long`

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

See https://s3.amazonaws.com/clickhouse-test-reports/66506/51a01fd57660742f3287dea9d9416f0620143233/stateless_tests__tsan__s3_storage__[2_4].html

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 7ab1664fe543e64680ac6bea866d9f2c8052ba1d with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67284/7ab1664fe543e64680ac6bea866d9f2c8052ba1d/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0fa6e1006bbeb5479558397160e4e46981aed741/unit_tests__release_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/01600_parts_states_metrics_long.sh [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
+# Tags: long
```
Comment (azat) [inline tests/queries/0_stateless/01600_parts_states_metrics_long.sh:2]: Maybe `no-parallel` will help?

File: tests/queries/0_stateless/01600_parts_states_metrics_long.sh [hunk 2]
```diff
@@ -15,17 +16,26 @@ verify_sql="SELECT
 # In case of test failure, this code will do infinite loop and timeout.
 verify()
 {
-    while true
+    for i in {1..5000}
     do
-        result=$( $CLICKHOUSE_CLIENT -m --query="$verify_sql" )
-        [ "$result" = "1" ] && break
+        result=$( $CLICKHOUSE_CLIENT --query="$verify_sql" )
```
Comment (azat) [inline tests/queries/0_stateless/01600_parts_states_metrics_long.sh:21]: You can remove original `verify_sql`

File: tests/queries/0_stateless/01600_parts_states_metrics_long.sh [hunk 3]
```diff
@@ -15,17 +16,26 @@ verify_sql="SELECT
...
+        [ "$result" = "1" ] && echo "$result" && break
         sleep 0.1
```
Comment (azat) [inline tests/queries/0_stateless/01600_parts_states_metrics_long.sh:23]: This sleep is too small for launching debug/sanitizer binaries btw

File: tests/queries/0_stateless/01600_parts_states_metrics_long.sh (hunk 1 above)
Comment (azat) [reply tests/queries/0_stateless/01600_parts_states_metrics_long.sh:2]: - https://github.com/ClickHouse/ClickHouse/pull/68136

**Pull request merged**